pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -427,6 +427,9 @@ protected boolean consumeLoop()
       try {
         messageBatch =
             _partitionGroupConsumer.fetchMessages(_currentOffset, null, _streamConfig.getFetchTimeoutMillis());
+        //track realtime rows fetched on a table level. This included valid + invalid rows
+        _serverMetrics.addMeteredTableValue(_clientId, ServerMeter.REALTIME_ROWS_FETCHED,
+            messageBatch.getMessageCount());
```

Comment:
I think this is not correct. It should be `messageBatch.getUnfilteredMessageCount()`.

Reply:
Thanks for pointing this out